actions/wait-for-workflow-job/action.yml

```diff
@@ -0,0 +1,137 @@
+name: Wait For Workflow Job
+description: |
+  Waits for a job in a workflow to complete
+  This is useful when you need to wait for a specific (required) job to complete
+  but do not need the result of the whole workflow itself.
+
+inputs:
+  repo:
+    description: Owner and repo in the form owner/repo
+    required: true
+  workflow-run-id:
+    description: The id of the workflow running the job
+    required: true
+  job-name:
+    description: The name of the job
+    required: true
+  github-token:
+    description: The github token to use
+    required: true
+  retry-delay:
+    description:
+      The amount of time to sleep between calls to check for job complete
+    required: false
+    default: "60"
+  timeout:
+    description:
+      How long in seconds to check for the job to start and complete before
+      timing out and giving up
+    required: false
+    default: "1800" # 30 minutes
+
+runs:
+  using: composite
+  steps:
+    - name: Wait For Workflow Job
+      shell: bash
+      id: get-run-id
+      env:
+        GH_TOKEN: ${{ inputs.github-token }}
+        JOB_NAME: ${{ inputs.job-name }}
+        REPO: ${{ inputs.repo }}
+        WF_RUN_ID: ${{ inputs.workflow-run-id }}
+        RETRY_DELAY: ${{ inputs.retry-delay }}
+        TIMEOUT: ${{ inputs.timeout }}
+      run: |
+        # Convert sleep_time to an integer if necessary (Bash typically handles this, but it's good practice)
+        sleep_time=$(($RETRY_DELAY + 0))
+        # Set a maximum wait time
+        max_wait_seconds=$(($TIMEOUT + 0))
+        start_time=$(date +%s)
+
+        # wait until the job starts and then grab its job id, or until we timeout
+        while :; do
+            current_time=$(date +%s) # Get the current time in seconds since epoch
+            elapsed_time=$((current_time - start_time))
+
+            # Check if the maximum wait time has been exceeded
+            if [ "$elapsed_time" -ge "$max_wait_seconds" ]; then
+                echo "Timeout reached without job starting."
+                exit 1
+            fi
+
+            # Fetch the list of jobs from the GitHub API
+            jobs=$(curl -H "Authorization: token $GH_TOKEN" \
+              -H "Accept: application/vnd.github.v3+json" \
+              "https://api.github.com/repos/${REPO}/actions/runs/$WF_RUN_ID/jobs?per_page=100")
```

Comment:
Same as above. Is there a reason for using curl?

Reply:
Now uses gh cli: https://github.com/smartcontractkit/.github/pull/454/commits/e02e6b59110bbf3ca6c14a23d1fa1641d2b0bde0